File: mifosng-android/src/main/java/com/mifos/objects/collectionsheet/CollectionSheet.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */

package com.mifos.objects.collectionsheet;

import java.util.List;

/**
 * Created by ishankhanna on 18/07/14.
 */
public class CollectionSheet {

    private List<Integer> dueDate;
    private LoadProducts loanProducts;
    private List<Groups> groups;
    private EntityType attendanceTypeOptions;

    public List<Integer> getDueDate() {
        return dueDate;
    }

    public void setDueDate(List<Integer> dueDate) {
        this.dueDate = dueDate;
    }

    public LoadProducts getLoanProducts() {
        return loanProducts;
    }

    public void setLoanProducts(LoadProducts loanProducts) {
        this.loanProducts = loanProducts;
    }

    public List<Groups> getGroups() {
        return groups;
    }

    public void setGroups(List<Groups> groups) {
        this.groups = groups;
    }

    public EntityType getAttendanceTypeOptions() {
        return attendanceTypeOptions;
    }

    public void setAttendanceTypeOptions(EntityType attendanceTypeOptions) {
        this.attendanceTypeOptions = attendanceTypeOptions;
    }

    @Override
    public String toString() {
        return "CollectionSheet{" +
                "dueDate=" + dueDate +
                ", loanProducts=" + loanProducts +
                ", groups=" + groups +
                ", attendanceTypeOptions=" + attendanceTypeOptions +
                '}';
    }
}

File: core/model/src/commonMain/kotlin/com/mifos/core/model/objects/databaseobjects/CollectionSheet.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.core.model.objects.databaseobjects

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json

@Serializable
data class CollectionSheet(
    var dueDate: IntArray,
    var groups: List<MifosGroup>,
) {
    override fun toString(): String {
        return Json.encodeToString(serializer(), this)
    }
}
